fl4health/tasks/autoencoder_trainer.py

```diff
@@ -0,0 +1,182 @@
+from pathlib import Path
+from typing import Callable, Dict, Tuple
+
+import torchvision.transforms as transforms
+from torch.nn.modules.loss import _Loss
+from torch.utils.data import DataLoader
+
+from fl4health.processing.VAE_dim_reduction import ClientConditionedProcessor, VAEProcessor
+from fl4health.processing.VAE_training import VAETransformer
+from fl4health.tasks.loss import VAE_loss
+from fl4health.utils.sampler import LabelBasedSampler
+
+
```

Comment:
I might be wrong, but one good thing about having a task module and these trainer classes is that if we decide to improve or change any of the code in processing, model-specific loss functions, or data transformers, users don't need to change their code.